### PR TITLE
Fixes #31648 - Add Content view filter details for package groups

### PR DESCRIPTION
**Pull request by johnpmitsch**
This will show package groups for a filter in the new CV details page e.g. `/labs/content_views/1#filters?subContentId=1`

TODO
- [x] small cleanup (marked as TODOs in the code)
- [x] finish tests

### CODE REVIEW

**Comment by johnpmitsch** [comment]
I'm finishing up the tests and some cleanup but it's ready for review

**Comment by johnpmitsch** [comment]
Here is a screenshot (cc @mari

**Comment by MariSvirik** [comment]
@johnpmitsch it looks good. Just a couple of things:

- Get rid of the icon in the 'View tasks' button
- Include label + icon + package group should align the layer to the middle, not to the top. 
(Not sure if I describe it precisely, ping me if necessary)

- Pagination should be also under the table as well. In mock I use [compact pagination](https://www.patternfly.org/v4/components/pagination/design-guidelines/#compact-pagination) due to space limitation after adding the rest of the toolbar. 

- [Bulk selector ](https://www.patternfly.org/v4/guidelines/bulk-selection/#bulk-selector)should be in a toolbar not in a table header.

- There should be 16 px between the edge of the search field/pagination and the table header.

**Review by jturel — COMMENTED** [review]
Since it's a draft I gave it a quick review with less of a focus on code organization and things like that. tried to keep it higher level. functionality looks good in my browser!

**Comment by johnpmitsch** [comment]
@jturel I have to fix a broken test, but otherwise ready for re-review!

File: webpack/scenes/ContentViews/Details/Filters/ContentViewFilterDetailsHeader.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    <>
+      <GridItem span={12}>
+        <TextContent>
+          <Text component={TextVariants.h2}>{name}</Text>
+        </TextContent>
+      </GridItem>
+      <GridItem span={10}>
+        <Split hasGutter>
+          <SplitItem>
+            <Label color="blue">{inclusion ? 'Include' : 'Exclude' }</Label>
```
Comment (jturel) [inline webpack/scenes/ContentViews/Details/Filters/ContentViewFilterDetailsHeader.js:25]: Translation needed here

File: webpack/scenes/ContentViews/Details/Filters/ContentViewFilterDetails.js [hunk 2]
```diff
@@ -1,13 +1,55 @@
...
+import useUrlParamsWithHash from '../../../../utils/useUrlParams';
+import Loading from '../../../../components/Loading';
+import ContentViewFilterDetailsHeader from './ContentViewFilterDetailsHeader';
+import CVPackageGroupFilterContent from './CVPackageGroupFilterContent';
 
 const ContentViewFilterDetails = () => {
-  const { id } = useParams();
-  const { hash } = useLocation();
-  const { params: { subContentId } } = paramsFromHash(hash);
+  // TODO remove default params for testing
```
Comment (jturel) [inline webpack/scenes/ContentViews/Details/Filters/ContentViewFilterDetails.js:18]: I think you've completed this TODO :)

**Review by jturel — COMMENTED** [review]
Behavior still looks good!

Aside from the two small comments I do see a few errors that seems like easy fixes in the browser console to this effect

```
react_devtools_backend.js:2430 Warning: Failed prop type: Invalid prop `cvId` of type `string` supplied to `CVPackageGroupFilterContent`, expected `number`.
    in CVPackageGroupFilterContent (created by ContentViewFilterDetails)
    in ContentViewFilterDetai
```

**Comment by johnpmitsch** [comment]
@jturel ty, tests and feedback updated!

**Comment by johnpmitsch** [comment]
> @johnpmitsch it looks good. Just a couple of things:
> 
> * Get rid of the icon in the 'View tasks' button
> * Include label + icon + package group should align the layer to the middle, not to the top.
>   (Not sure if I describe it precisely, ping me if necessary)
> * Pagination should be also under the table as well. In mock I use [compact pagination](https://www.patternfly.org/v4/components/pagination/design-guidelines/#compact-pagination) due to space limitation after adding the rest of the toolbar.
> * [Bulk selector ](https://www.patternfly.org/v4/guidelines/bulk-selection/#bulk-selector)should be in a toolbar not in a table header.
> * There should be 16 px between the edge of the search field/pagination and the table header.
> * Page header should have a component cv icon (single cube) on the right side
> * Filter name and description should be editable

Thanks, a lot of this is being addressed outside of this PR and has open issues, I will make an issue to follow up with the editable filter name and description, and look at the alignment issue too.

**Review by jturel — APPROVED** [review]
ACK!

**Comment by johnpmitsch** [comment]
> Thanks, a lot of this is being addressed outside of this PR and has open issues, I will make an issue to follow up with the editable filter name and description, and look at the alignment issue too.

Filed here: https://projects.theforeman.org/issues/32038

**Pull request merged**